crates/component-macro/src/lib.rs

```diff
@@ -230,11 +396,138 @@ impl Expander for LiftExpander {
 
         Ok(expanded)
     }
+
+    fn expand_variant(
+        &self,
+        input: &DeriveInput,
+        cases: &[VariantCase],
+        _style: VariantStyle,
+    ) -> Result<TokenStream> {
+        let internal = quote!(wasmtime::component::__internal);
+
+        let mut lifts = TokenStream::new();
+        let mut loads = TokenStream::new();
+
+        for (index, VariantCase { ident, ty, .. }) in cases.iter().enumerate() {
+            let index_u8 = u8::try_from(index).map_err(|_| {
+                Error::new(
+                    input.ident.span(),
+                    "`enum`s with more than 256 variants not yet supported",
+                )
+            })?;
```

Comment:
Is this mostly to just simplify this for now where you don't have to worry about 16-vs-8  bit loads?

Reply:
Yeah, I was just keeping it simple to start with.  I was also not sure what the theoretical max was for the component model.  2^32?  More?

Reply:
The [Canonical ABI](https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md#alignment) defines that in the `discriminant_type` function, and the current maximum is 2^32.